pkg/oidcclient/login.go

```diff
@@ -354,12 +354,10 @@ func Login(issuer string, clientID string, opts ...Option) (*oidctypes.Token, er
 func (h *handlerState) needRFC8693TokenExchange(token *oidctypes.Token) bool {
 	// Need a new ID token if there is a requested audience value and any of the following are true...
 	return h.requestedAudience != "" &&
-		// we don't have an ID token
+		// we don't have an ID token (maybe it expired or was otherwise removed from the session cache)
```

Comment:
So `idTokenExpiredOrCloseToExpiring(token.IDToken)` ~ `...minstake would cause the RFC8693 token exchange to happen when the OIDC provider is not a Pinniped Supervisor, which would fail because most other providers do not support that type of token exchange.`

[RFC8693 being Oauth 2.0 Token Exchange](https://datatracker.ietf.org/doc/html/rfc8693#name-introduction).  Giving this a read to see if I can understand why non-Supervisor clients would likely not support this.

Reply:
It's not commonly needed and not part of the core spec.